integration/truncindex_test.go

```diff
@@ -0,0 +1,127 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package integration
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	runtimeapi "k8s.io/kubernetes/pkg/kubelet/apis/cri/v1alpha1/runtime"
+)
+
+func genTruncIndex(normalName string) string {
+	return normalName[:3]
+}
+
+func TestTruncIndex(t *testing.T) {
+	t.Logf("Pull an image")
+	const appImage = "busybox"
+	img, err := imageService.PullImage(&runtimeapi.ImageSpec{Image: appImage}, nil)
+	require.NoError(t, err)
+	// skip the prifix "sha256:"
+	imgTruncId := img[:len(img)/2]
+	defer func() {
+		assert.NoError(t, imageService.RemoveImage(&runtimeapi.ImageSpec{Image: imgTruncId}))
+	}()
+
+	t.Logf("Get image status by truncindex, truncId: %s", imgTruncId)
+	res, err := imageService.ImageStatus(&runtimeapi.ImageSpec{Image: imgTruncId})
+	require.NoError(t, err)
+	require.Equal(t, true, (res != nil) && (res.Id != ""))
+	t.Logf("Image status: %v", res)
+
```

Comment:
Please add a todo for failure test case where there are two images with the same truncindex.. if you add n images at least two will share the same leading digit.  "sha256:n" where n is the a number from 0-9 where two images have the same trunc, for example `sha256:9`

Reply:
@mikebrow 
There are some probability, but I have not good thoughts for handling it. Do you give me some suggestion?
Do you think that we do it in unit test of ImageStore? 
e.g.   in cri-containerd/pkg/store/image/image_test.go

Reply:
The test is fine for now.. just need to add a todo to cover adding 11 different images to force having two with the same leading x digit. Then loop on the images getting status. I was thinking just do it here, but I suppose you could also do it in the image store unit test. :-)

Reply:
done

Reply:
We just need to make sure an error is returned, if the trunc index is too short, and causes ambiguity.

Reply:
Now we can do it. Now there is sha256 prefix in truncindex.

Reply:
We cann't do it. When we use sha256 prefix, digestSet handle it as invalidFormat and imageStore.Get return store.ErrNotExist.